### ISSUE TITLE
Aborted connection 512 to db: 'cactidb' user: 'cactiuser' host: 'localhost' (Got an error reading communication packets)

### ISSUE THREAD

**Issue by lfrangulian** [Cacti/cacti#4270]
![cactiiii](https://user-images.githubusercontent.com/84389134/118654565-be288b80-b7f9-11eb-8c2a-5d82fafbd610.jpg)
Hello,

I have upgraded 0.8.7e to 1.2.15 for a while there was no issues, after adding new graphs i have started to receive the below error:

[Warning] Aborted connection 512 to db: 'cactidb' user: 'cactiuser' host: 'localhost' (Got an error reading communication packets)
[Warning] Aborted connection 457 to db: 'cactidb' user: 'cactiuser' host: 'localhost' (Got timeout reading communication packets)

I did remove and recrate the cactiuser, same problem.

I have looked everywhere and found nothing. last thing I did was upgrade from 1.2.15 to 1.2.17 and still the same.

Using CentOS Linux 7 and MariaDB 10.5.10

Any ideas ? how to fix it or what is causing this problem ? 

Regards,
Leon

**Comment by netniV** [comment]
I would check your MariaDB / MySQL error logs, if there is a problem reading packets either there is a MySQL or network problem (if the server is on a separate server - which I doubt as it says localhost).

**Comment by lfrangulian** [comment]
> I would check your MariaDB / MySQL error logs, if there is a problem reading packets either there is a MySQL or network problem (if the server is on a separate server - which I doubt as it says localhost).


MariaDB logs are only about the connection issue. The logs appear only if u go to tree and started to view graphs and refresh them. 
Can't find any other error logs in DB, and for network you are right, it's not on a separate server. I just run out if ideas how to solve this problem. There will be too many aborted connections and in the end the attached error will show up and connection to the DB is lost.

**Comment by TheWitness** [comment]
From my perspective, I've been seeing aborted connections more and more with MariaDB, but it could simply be that they changed the default logging for aborted connections.  I frankly turned off logging of aborted connections.  They are 100% coming from the Web Server, and don't really impact Cacti at all.  PDO is supposed to automatically close connections on php session closes, but I don't know about Apache's httpd or php-fpm service.  So, my guess it might be something in Apache/php-fpm.

**Comment by TheWitness** [comment]
To stop the warnings in the mariadb.log, add the following to your `server.cnf`

```
log_warnings = 0
```

**Comment by TheWitness** [comment]
The 500 error should be logged into the cacti.log.  If you can find it, post it here.  But the aborted connections are kind of a red herring.

**Comment by TheWitness** [comment]
One other note was that MariaDB had to issue a big time fix some time ago as there was a protocol error from older clients that were causing MariaDB to drop the connections.  They backed out the change, but they may have not backed it out entirely.  Certain older PHP PDO API's may still have errors, but I have not spent much time on tracing it.

**Comment by TheWitness** [comment]
I'm going to work off your hunch related to viewing/refreshing graphs.  I reset counters last night, and this morning there are no aborted connections.  I'm also using MariaDB 10.5.10, with about 60 devices and a 30 second data collection frequency.

Before I reset the counters, I did have 974 aborted connections, but this morning, there are not any yet.  I've set the auto-graph refresh frequency to 15 seconds, and will tool the interface to see if there is indeed a correlation.

What is interesting, is that I have multiple data collectors, and when running a 'show processlist' in MariaDB, I see some long lived connections from the remote data collector that are sleeping for 25k seconds.  Those will eventually become aborted.  Soon now in fact.
